### PR TITLE
Add local file connector

### PR DESCRIPTION
**Pull request by nileema**

### CODE REVIEW

**Comment by yuananf** [comment]
Is this only used to analyze presto's "http-request.log"?

File: presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileHandleResolver.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    @Override
+    public boolean canHandle(ConnectorSplit split)
+    {
+        return split instanceof LocalFileSplit && ((LocalFileSplit) split).getConnectorId().equals(connectorId);
+    }
+
+    @Override
+    public boolean canHandle(ConnectorTableLayoutHandle handle)
+    {
+        return handle instanceof LocalFileTableLayoutHandle;
```
Comment (electrum) [inline presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileHandleResolver.java]: This needs to check the connector ID, like the others

Comment (electrum) [inline presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileHandleResolver.java]: I think all the other connectors that don't do this are wrong

File: presto-server/src/main/provisio/presto.xml [hunk 2]
```diff
@@ -115,4 +115,11 @@
             <unpack />
         </artifact>
     </artifactSet>
+
+    <artifactSet to="plugin/localfile">
+        <artifact id="${project.groupId}:presto-local-file:zip:${project.version}">
+            <unpack />
+        </artifact>
+    </artifactSet>
+
```
Comment (electrum) [inline presto-server/src/main/provisio/presto.xml]: Extra blank line

File: presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileMetadata.java [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+                columns.put(tableName, getColumns(tableHandle));
+            }
+        }
+        return columns.build();
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        throw new UnsupportedOperationException();
```
Comment (electrum) [inline presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileMetadata.java]: Remove all these methods -- they are default methods in the interface which throw the correct exception type

**Comment by nileema** [comment]
Superseded by #5078

**Pull request closed without merging**